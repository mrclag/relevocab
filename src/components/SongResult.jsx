import React from 'react';

const SongResult = ({ name }) => {
  return (
    <div className="results-result">
      <div className="results-title">{name}</div>
      <div className="results-artist">Artistname</div>
      <div className="results-cards">
        <span className="card">word</span>
        <span className="card">word</span>
        <span className="card">word</span>
        <span className="card">word</span>
        <span className="card">word</span>
      </div>
    </div>
  );
};

export default SongResult;
