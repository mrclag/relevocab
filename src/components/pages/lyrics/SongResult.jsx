import React from 'react';

const SongResult = ({ name, data }) => {
  return (
    <div className="results-result">
      <div className="results-title">{data.title}</div>
      <div className="results-artist">{data.artist}</div>
      <div className="results-cards">
        {Object.keys(data.cards).map((cardId, i) => (
          <span key={i} className="card">
            {data.cards[cardId].front}
          </span>
        ))}
      </div>
    </div>
  );
};

export default SongResult;
