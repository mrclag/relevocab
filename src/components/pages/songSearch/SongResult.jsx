import React from 'react';
import { SongResultWrapper } from './SongResult.styles.js';

const SongResult = ({ song }) => {
  console.log(song);
  return (
    <SongResultWrapper>
      <img src={song.imgUrl} className="deck-image" alt="Deck"></img>
      <div className="deck-info">
        <div className="deck-title">
          {song.artist} - {song.title}
        </div>
        <div className="deck-cards">{song.cards.length} cards</div>
        <div className="deck-buttons">
          <button className="practice">Add Song</button>
          <button className="details">Details</button>
        </div>
      </div>
    </SongResultWrapper>
  );
};

export default SongResult;
