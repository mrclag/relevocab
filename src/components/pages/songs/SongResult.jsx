import React from 'react';
import { connect } from 'react-redux';

import { SongResultWrapper } from './SongResult.styles';
import SongLine from './SongLine';
import { addSongAsDeck } from '../../../store/actions/deckActions';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlusSquare } from '@fortawesome/free-solid-svg-icons';

export const SongResult = ({ song, addSongAsDeck }) => {
  const addSong = (song) => {
    console.log('running add song on ', song);
    addSongAsDeck(song);
  };

  return (
    song && (
      <SongResultWrapper>
        <div className="song-img" onClick={() => addSong(song)}>
          <img src={song.imgUrl} alt="" />
          <div className="overlay">
            <FontAwesomeIcon
              icon={faPlusSquare}
              className="icon"
              style={{ fontSize: '40px' }}
            />
          </div>
        </div>
        <div className="song-info">
          <div className="title">{song.title}</div>
          <div className="artist">{song.artist}</div>
          <div className="numcards">{song.cards.length} cards</div>
        </div>
        <div className="song-cards">
          {song.cards.slice(0, 7).map((card, i) => (
            <SongLine key={i} card={card} />
          ))}
          <SongLine
            card={{
              front: `${song.cards.length - 6} more...`,
              back: `${song.cards.length - 6} more...`,
            }}
          />
        </div>
      </SongResultWrapper>
    )
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    addSongAsDeck: (song) => dispatch(addSongAsDeck(song)),
  };
};

export default connect(null, mapDispatchToProps)(SongResult);
