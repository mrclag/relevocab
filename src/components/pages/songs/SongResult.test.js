import React from 'react';
import { SongResult } from './SongResult.jsx';
import { shallow } from 'enzyme';

describe('SongResult', () => {
  const songResult = shallow(<SongResult />);

  it('renders correctly', () => {
    expect(songResult).toMatchSnapshot();
  });
});
