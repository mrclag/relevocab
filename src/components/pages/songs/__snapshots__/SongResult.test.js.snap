// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongResult renders correctly 1`] = `ShallowWrapper {}`;
